agent/consul/leader_metrics.go

```diff
@@ -151,3 +152,17 @@ func initLeaderMetrics() {
 		metrics.SetGaugeWithLabels(g.Name, float32(math.NaN()), g.ConstLabels)
 	}
 }
+
+// shouldWarn checks to see if we are close enough to the cert expiring that we
+// should send out a WARN log message.
+// It defaults to returning true if the cert will expire within 28 days or 40%
+// of the certs total duration, whichever is shorter.
+func expiresSoon(before, after time.Duration) bool {
+	warningPeriod := 28 * (24 * time.Hour) // 28 days (default)
+	seventyDays := 70 * (24 * time.Hour)   // 28 days is 40% of 70 days
+	totalDuration := time.Duration(before + after)
```

Comment:
I had a hard time following this logic because I didn't realize that `before` was actually `timeSinceLifetimeStart` and `after` is actually `timeUntilLifetimeEnd`, which together are equal to `totalLifetime`. Renaming the arguments might help.

Reply:
They are named NotBefore and NotAfter in the Cert itself, which is where I derived the name from. I could expand the names to add the not to make that more clear. That naming should be very familiar to anyone who works with Certs.

Reply:
To expand a bit.. The names previously used are not good. They are too long and, importantly, don't reflect the field name in the Cert that they represent. I do think adding the not will help though and will do that.

Reply:
I agree that `NotBefore` and `NotAfter` are clear names. That said, they don't reflect the value of what's being passed in. Both of those fields are timestamps, whereas what's being passed into this function are durations relative to the current time.

For example, when I read the test case that set `before: (day * 50), after: (day * 20),` I thought it was conveying a cert that was always expired (because `NotBefore > NotAfter`). But in actuality, it's intending to express a cert that is 50/70 through its lifetime.

Reply:
TLDR; How about NotAfter and Lifetime as the 2 variables instead?

I get what you mean but IMO the importance of maintaining coherence with the Certificate names is more important. An alternative would be to change the Query's signature to carry the total time and the NotAfter time. NotAfter is required for the checks but NotBefore isn't, it is just a convenient way to get the missing data to calculate the total lifetime (as we already have NotAfter) but the relevant bit is the total duration not the start time.

Reply:
Trying out the new names.. it does make the tests more readable for sure.

Reply:
I like the new names and have pushed them up. Let me know if you think this helps with the readability. Thanks!